apps/system/js/app_transition_controller.js

```diff
@@ -292,9 +292,19 @@
     // See: https://bugzilla.mozilla.org/show_bug.cgi?id=961557
     // XXX: We should let HierarchyManager to manage the focus.
     // See: https://bugzilla.mozilla.org/show_bug.cgi?id=1079748
+    var bottomWindowType = this.app.getBottomMostWindow().CLASS_NAME;
+    if (bottomWindowType === 'HomescreenWindow') {
+      bottomWindowType = 'AppWindow';
+    }
     return (this._transitionState == 'opened' &&
-            !rocketbar.active &&
-            !Service.query('SimLockManager.isActive'));
+            Service.query('getTopMostWindow') === this.app &&
+            Service.query('getTopMostUI').name.indexOf(bottomWindowType) >= 0);
+    // Possible bottom most windows:
+    // AppWindow/HomscreenWindow, AttentionWindow, SecureWindow,
+    // LockScreenWindow
+    // which belong to
+    // AppWindowManager, AttentionWindowManager, SecureWindowManager,
+    // LockScreenWindowManager.
```

Comment:
Ok. This makes sense but I'd like to see if we can come up with a stronger alternative than an `str.startsWith`...

* maybe having a `const` mapping `CLASS_NAME`s to hierarchy UI names
* or giving each window instance a reference to its `.manager` (but I could see why we don't want to go there...)

What do you think?